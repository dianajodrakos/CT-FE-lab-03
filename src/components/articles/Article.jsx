import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ author, title, description, url, image }) => (
  <a href={url} rel="noreferrer" target="_blank">
    <figure>
      <img src={image} alt={title} />
    </figure>
    <h2>{title}</h2>
    <h3>{author}</h3>
    <p>{description}</p>
  </a>
);

Article.propTypes = {
  author: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Article;
