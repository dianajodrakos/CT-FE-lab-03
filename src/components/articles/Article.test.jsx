import React from 'react';
import { render, screen } from '@testing-library/react';
import Article from './Article';

describe('Article', () => {
  it('displays an article', () => {
    render(
      <Article 
        author="author"
        title="title"
        description="article description"
        url="http://www.google.com"
        image="https://via.placeholder.com/300x200.png?text=Test+Image"
        published="timestamp"
      />
    );

    const article = screen.getByText('article description');
    expect(article).toMatchSnapshot();
  });
});
