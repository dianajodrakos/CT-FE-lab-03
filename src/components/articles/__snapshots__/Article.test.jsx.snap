// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article displays an article 1`] = `
<p>
  article description
</p>
`;
